### PR TITLE
fix(appversion): fix for wrong version parsing

### PR DESCRIPTION
**Pull request by gal-yardeni**
we actually need the artifact name to parse the version with Frigga.

### CODE REVIEW

File: keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt [hunk 1]
```diff
@@ -36,14 +36,6 @@ class DebianArtifactSupplier(
   override val supportedVersioningStrategy =
     SupportedVersioningStrategy("deb", DebianVersioningStrategy::class.java)
 
-  override fun publishArtifact(artifact: PublishedArtifact) {
```
Comment (gal-yardeni) [inline keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt:39]: we don't need this (it's duplicated to the `shouldProcess` logic below).

**Review by emjburns — APPROVED** [review]

File: keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt (hunk 1 above)
Comment (luispollo) [reply keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt:39]: This is pretty surprising to me, given the description of the PR. This is the function that gets called by `ArtifactController` to publish a new artifact. Has the base class been modified to call `shouldProcessArtifact`?

File: keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt [hunk 2]
```diff
@@ -174,6 +182,11 @@ internal class DebianArtifactSupplierTests : JUnit5Minutests {
           .isTrue()
       }
 
+      test ("should process artifact successfully2") {
```
Comment (luispollo) [inline keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt:185]:
```suggestion
      test ("should process artifact successfully") {
```

Comment (gal-yardeni) [reply keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt:185]: I have 2 tests with the same name, so this is 2 😄  will fix later on

File: keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt [hunk 3]
```diff
@@ -52,6 +52,14 @@ internal class DebianArtifactSupplierTests : JUnit5Minutests {
       metadata = mapOf("releaseStatus" to SNAPSHOT, "buildNumber" to "1", "commitId" to "a15p0")
     )
 
+    val actualVersionInPublishedArtifact = PublishedArtifact(
```
Comment (luispollo) [inline keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt:55]: If this is the actual format of a Debian published artifact (i.e. without the name part in the `version` string), then I think a better approach would be to modify `latestArtifact` above and just keep the existing test.

Comment (gal-yardeni) [reply keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplierTests.kt:55]: yeah good point.

File: keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt (hunk 1 above)
Comment (gal-yardeni) [reply keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt:39]: `artifactListener` is calling this. It was in previous PRs.

**Pull request merged**